server/test/endpoint_roles.test.js

```diff
@@ -0,0 +1,33 @@
+const { describe, it } = require('mocha')
+const fetch = require('node-fetch')
+const assert = require('assert')
+const { getAdmin1IdPToken } = require('./lib/tokens')
+const { getServerUrl } = require('./lib/urls')
+const dbAdminRole = require('database/layer/admin-client')
+
+describe('Admin Roles API Tests', () => {
+  const url = `${getServerUrl()}`
+  let requestHeaders
+  before(async function () {
+    // GET IDENTITY-PROVIDER TOKEN
+    const userAccessToken = await getAdmin1IdPToken()
+    requestHeaders = {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${userAccessToken}`
+    }
+  })
+
+  describe('Create Roles', () => {
+    it('Should create a new role')
+    it('Should return a 400 for a duplicate role within a client')
+    it('Should return a 400 for invalid client_id', async () => {
```

Comment:
Is it bad request because `tenant_name` is missing? why is the `client_id` invalid? I believe `foo` passes the regex check `^[a-z0-9-_]*$`.

Reply:
@k1ch  this is a bad request because the client id `foo` does not exist. This test is trying to create a new Role, so it needs a valid client id to associate the Role with.